PerfettoCds/Pipeline/CompositeDataCookers/PerfettoGenericEventCooker.cs

```diff
@@ -230,20 +234,37 @@ public void OnDataAvailable(IDataExtensionRetrieval requiredData)
                     processName = $"{result.process.Name} {result.process.Pid}";
                 }
 
+                int parentTreeDepthLevel = 0;
+                long? currentParentId = result.slice.ParentId;
+                
+                // Walk the parent tree
+                while (currentParentId.HasValue)
+                {
+                    var parentPerfettoSliceEvent = sliceData[(int) currentParentId.Value];
+                    Debug.Assert(parentPerfettoSliceEvent == null || (parentPerfettoSliceEvent.Id == currentParentId.Value)); // Since we relying on index being the Id
```

Comment:
Maybe we should change the slice query to be "order by id" instead of "order by ts", just to be safe

Reply:
Yea, good thinking